### ISSUE TITLE
Client Side Validation Generic function

### ISSUE THREAD

**Issue by frinzekt** [uwasystemhealth/IndEAA#25]
`handleChange` function usually follows a very similar pattern, see my implementation for the Living Lab Contact Page (Contact Form Modal component).

Having a generic function to do this will help very much in any form related client side validation.

See the state structure and handleChange Validation function
https://github.com/uwasystemhealth/livinglab-site-v3/blob/master/components/Context/index.js
tldr: There are 2 types of states for each form. The first state contains the values of the field in the form, and the second state contains all the error message of the specific field.. The `handleChange` function determines the form field based on name. With that, it uses a switch statement to determine which validator function it should use. The `validateModifyState` function takes 3 parameters: 

1. the value of the form field
2. the validator function it will use (will it use for email, for numbers or whatever)
3. The error message to display when the value did not pass the validator function

https://github.com/uwasystemhealth/livinglab-site-v3/blob/master/helpers/validation.js
tldr: regex

See example of how to render the error in the forms
https://github.com/uwasystemhealth/livinglab-site-v3/blob/master/components/Layout/Contact.js
tldr: you can customise the MaterialKit Component to your own component to make it better. The `fieldName` should be the same as the field name in the state or something. This is because I customised the `handleChange` function to determine which form field is to which field in the state (which means I only really have to have 1 handleChange function for all the field`
